### PR TITLE
Support for PReLU with multiple dims slope tensor for GPU

### PR DESCRIPTION
**Pull request by e-ddykim**
### Details:
 - PReLU with multiple dims slope tensor fails in the gpu plugin.
 - This PR solves this issue by reshaping the slope tensor according to the numpy broadcasting rule.

### Tickets:
 - 84753, 84798

### CODE REVIEW

File: src/plugins/intel_gpu/src/plugin/ops/constant.cpp [hunk 1]
```diff
@@ -115,6 +115,14 @@ static void CreateConstantOp(Program& p, const std::shared_ptr<ngraph::op::v0::C
             handleConvWeights(outOp, consts, numConstUsers, false);
         } else if (ngraph::is_type<ngraph::op::v1::GroupConvolutionBackpropData>(outOp) && node.get_index() == 1) {
             handleConvWeights(outOp, consts, numConstUsers, true);
+        } else if (ngraph::is_type<ngraph::op::v0::PRelu>(outOp) && node.get_index() == 1) {
+            auto input_shape = outOp->get_input_shape(0);
+            if (constDims.size() != 1 && constDims.size() < input_shape.size()) {
+                ngraph::Shape slope_shape(input_shape.size(), 1);
+                for (int j = 1; j <= constDims.size(); j++)
+                    slope_shape[slope_shape.size()-j] = constDims[constDims.size()-j];
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/plugin/ops/constant.cpp:123]: Was this passing the lint? Please adjust the blanks

File: src/plugins/intel_gpu/src/plugin/ops/unary.cpp [hunk 2]
```diff
@@ -85,7 +85,7 @@ static void CreatePReluOp(Program& p, const std::shared_ptr<ngraph::op::v0::PRel
         if (!ngraph::op::util::get_single_value(slope_node, slope))
             IE_THROW() << "Unsupported parameter size in " << op->get_friendly_name() << " (" << op->get_type_name() << ")";
         CreateUnaryEltwiseOp(p, op, cldnn::activation_func::relu_negative_slope, {slope});
-    } else if (out_shape.size() >= 2 && ngraph::shape_size(slope_shape) == out_shape[1]) {
```
Comment (yeonbok) [inline src/plugins/intel_gpu/src/plugin/ops/unary.cpp:88]: Could you explain why the second condition removed?

Comment (e-ddykim) [reply src/plugins/intel_gpu/src/plugin/ops/unary.cpp:88]: For example, in the case where the shape of the slope tensor is [1, N, M, 1], `ngraph::shape_size(slope_shape)` is `N x M` and `out_shape[1]` is `N`. So I remove the second condition.

**Comment by yeonbok** [comment]
I am little bit confused becuase
1. In topology creation (src/plugin/ops/constant.cpp), the constant was reshaped  to have slop only in the feature dim e.g., [1, N, 1, 1] . It sounds like that the "cldnn topology" can only handle such shape. 
2. However the cldnn unittest with topology creaetd wih 2 dim slopes can pass the test..right? (though they are not going through the reshape in src/plugin/ops/constant.cpp) 

Could you summarize the final policy? How should the slope dimension should be handled in cldnn? 
Should the constant be reshaped to [1, N, 1, 1] ? Or should it not?

**Comment by e-ddykim** [comment]
> I am little bit confused becuase
> 
> 1. In topology creation (src/plugin/ops/constant.cpp), the constant was reshaped  to have slop only in the feature dim e.g., [1, N, 1, 1] . It sounds like that the "cldnn topology" can only handle such shape.
> 2. However the cldnn unittest with topology creaetd wih 2 dim slopes can pass the test..right? (though they are not going through the reshape in src/plugin/ops/constant.cpp)
> 
> Could you summarize the final policy? How should the slope dimension should be handled in cldnn? Should the constant be reshaped to [1, N, 1, 1] ? Or should it not?

Okay, let me summarize the final policy.
There are two cases for the PReLU slope shapes, one dimension or not.

1. 1-dim slope is handled by `getConstTensor` @ https://github.com/e-ddykim/openvino/blob/3b855abd9992642e7d7a48d122c9544d7ab4ab4f/src/plugins/intel_gpu/src/plugin/ops/constant.cpp#L45
[1] --> [1, 1, 1, 1]
[N] --> [1, N, 1, 1]

2. multi-dim slope is handled by numpy broadcasting rule (right-aligned) @ https://github.com/e-ddykim/openvino/blob/3b855abd9992642e7d7a48d122c9544d7ab4ab4f/src/plugins/intel_gpu/src/plugin/ops/constant.cpp#L120
[N, 1, 1] --> [1, N, 1, 1]
[N, M, 1] --> [1, N, M, 1]

**Comment by yeonbok** [comment]
> > 1. In topology creation (src/plugin/ops/constant.cpp), the constant was reshaped  to have slop only in the feature dim e.g., [1, N, 1, 1] . It sounds like that the "cldnn topology" can only handle such shape.
> > 2. However the cldnn unittest with topology creaetd wih 2 dim slopes can pass the test..right? (though they are not going through the reshape in src/plugin/ops/constant.cpp)
> > 
> > Could you summarize the final policy? How should the slope dimension should be handled in cldnn? Should the constant be reshaped to [1, N, 1, 1] ? Or should it not?
> 
> Okay, let me summarize the final policy. There are two cases for the PReLU slope shapes, one dimension or not.
> 
> 1. 1-dim slope is handled by `getConstTensor` @ https://github.com/e-ddykim/openvino/blob/3b855abd9992642e7d7a48d122c9544d7ab4ab4f/src/plugins/intel_gpu/src/plugin/ops/constant.cpp#L45
>    [1] --> [1, 1, 1, 1]
>    [N] --> [1, N, 1, 1]
> 2. multi-dim slope is handled by numpy broadcasting rule (right-aligned) @ https://github.com/e-ddykim/openvino/blob/3b855abd9992642e7d7a48d122c9544d7ab4ab4f/src/plugins/intel_gpu/src/plugin/ops/constant.cpp#L120
>    [N, 1, 1] --> [1, N, 1, 1]
>    [N, M, 1] --> [1, N, M, 1]

Ah I thought that the constant is reshaped to [1, N, 1, 1] from our old conversation. 
But as you mentioned, now it can accept [1, N, M, 1]
And cldnn's input slop can have multiple dim slope as long as they are following the above broadcasting rule. 
Could you add that policy as a comment at your changed code in the constant.cpp?

File: src/plugins/intel_gpu/src/plugin/ops/constant.cpp (hunk 1 above)
Comment (e-ddykim) [reply src/plugins/intel_gpu/src/plugin/ops/constant.cpp]: It passed the lint. But I added blanks between words for better readability.

**Comment by e-ddykim** [comment]
> > > 1. In topology creation (src/plugin/ops/constant.cpp), the constant was reshaped  to have slop only in the feature dim e.g., [1, N, 1, 1] . It sounds like that the "cldnn topology" can only handle such shape.
> > > 2. However the cldnn unittest with topology creaetd wih 2 dim slopes can pass the test..right? (though they are not going through the reshape in src/plugin/ops/constant.cpp)
> > > 
> > > Could you summarize the final policy? How should the slope dimension should be handled in cldnn? Should the constant be reshaped to [1, N, 1, 1] ? Or should it not?
> > 
> > 
> > Okay, let me summarize the final policy. There are two cases for the PReLU slope shapes, one dimension or not.
> > 
> > 1. 1-dim slope is handled by `getConstTensor` @ https://github.com/e-ddykim/openvino/blob/3b855abd9992642e7d7a48d122c9544d7ab4ab4f/src/plugins/intel_gpu/src/plugin/ops/constant.cpp#L45
> >    [1] --> [1, 1, 1, 1]
> >    [N] --> [1, N, 1, 1]
> > 2. multi-dim slope is handled by numpy broadcasting rule (right-aligned) @ https://github.com/e-ddykim/openvino/blob/3b855abd9992642e7d7a48d122c9544d7ab4ab4f/src/plugins/intel_gpu/src/plugin/ops/constant.cpp#L120
> >    [N, 1, 1] --> [1, N, 1, 1]
> >    [N, M, 1] --> [1, N, M, 1]
> 
> Ah I thought that the constant is reshaped to [1, N, 1, 1] from our old conversation. But as you mentioned, now it can accept [1, N, M, 1] And cldnn's input slop can have multiple dim slope as long as they are following the above broadcasting rule. Could you add that policy as a comment at your changed code in the constant.cpp?

I added comments as you guided. Thank you.

File: src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp [hunk 3]
```diff
@@ -662,6 +762,28 @@ TEST_P(conv_fp32_prelu_eltwise, vector_ops_mixed_types) {
...
+        reorder("reorder_bfyx", "eltwise", p.default_format, data_types::f32)
+    );
+
+    implementation_desc conv_impl = { format::b_fs_yx_fsv16, "" };
+    bo_fused.set_option(build_option::force_implementations({ { "conv_prim", conv_impl } }));
+
+    tolerance = 1e-5f;
+    execute(p);
+}
+
```
[the remark this answers is not in the document]
Comment (e-ddykim) [reply src/plugins/intel_gpu/tests/fusions/convolution_fusion_test.cpp:786]: 1) I tested it with the target network on DG2, and it passed the test.

2) When these test cases ran through onednn, some test cases failed. Since the failed test cases fail without this PR, I don't think that this PR is the problem.
